src/reflect/scala/reflect/internal/Types.scala

```diff
@@ -4485,15 +4485,22 @@ trait Types
 
   /** Do type arguments `targs` conform to formal parameters `tparams`?
    */
-  def isWithinBounds(pre: Type, owner: Symbol, tparams: List[Symbol], targs: List[Type]): Boolean = {
-    var bounds = instantiatedBounds(pre, owner, tparams, targs)
-    if (targs exists typeHasAnnotations)
+  def isWithinBounds(pre: Type, owner: Symbol, tparams: List[Symbol], targs: List[Type]): Boolean =
+    if (targs exists typeHasAnnotations){
+      var bounds = instantiatedBounds(pre, owner, tparams, targs)
       bounds = adaptBoundsToAnnotations(bounds, tparams, targs)
-    (bounds corresponds targs)(boundsContainType)
-  }
+      (bounds corresponds targs)(boundsContainType)
+    } else
+      (tparams corresponds targs){ (tparam, targ) =>
+        val bound = instantiatedBound(pre, owner, tparams, targs, tparam)
+        boundsContainType(bound, targ)
+      }
 
   def instantiatedBounds(pre: Type, owner: Symbol, tparams: List[Symbol], targs: List[Type]): List[TypeBounds] =
-    mapList(tparams)(_.info.asSeenFrom(pre, owner).instantiateTypeParams(tparams, targs).bounds)
+    mapList(tparams)(instantiatedBound(pre, owner, tparams, targs, _))
+
+  private[this] def instantiatedBound(pre: Type, owner: Symbol, tparams: List[Symbol], targs: List[Type], tparam: Symbol): TypeBounds =
+    tparam.info.asSeenFrom(pre, owner).instantiateTypeParams(tparams, targs).bounds
```

Comment:
They definitely want to be local, especially because of all the args.

Reply:
@som-snytt I have made them localised, and remove the non-api method `instantiatedBounds` that was only called from this file.